src/highlight.js

```diff
@@ -917,6 +922,24 @@ https://highlightjs.org/
     return lang && !lang.disableAutodetect;
   }
 
+  function addPlugin(plugin, options) {
+    plugins.push(plugin);
+  }
+
+  function fire(event, args) {
+    var cb = eventToFuncName(event);
+    plugins.forEach(function (plugin) {
+      if (plugin[cb]) {
+        plugin[cb](args);
+      }
+    });
+  }
+
+
+  function eventToFuncName(event) {
```

Comment:
What is the purpose of having slightly different names for events and functions?

Reply:
Easier to write, easier to read, and it's a convention i've seen used in other places.  It's just a nicety.

Reply:
In "Callbacks" section you give the names of the events. And in the actual code (according to examples) the plugin author must use function names. How to match one with the other?

Reply:
I changed the first example. :-)  I'm still thinking about the class based approach.  This would be much simpler in Ruby. :-)

Honestly it doesn't bother me to have/support both though... depending on how you write the plugin.

Reply:
> I changed the first example

Ah, I see. What I can't see, is why it is better than simply `afterHighlightBlock` (:

>  we'd likely have a section explaining how the event names map to function names, which I think is pretty straight forward to understand

Perhaps, but still I can't see benefits of having this abstraction. I don't insist on 1:1 mapping, just trying to understand what it might be useful for.

Reply:
I was hasty about the first example. It makes sense, because in that case we don't bother with mapping, we just specify the event name, which is good.

But class-based approach doesn't fit. Of course you can instantiate an object of your class and register its methods as callbacks...

Reply:
```
  class DataLanguagePlugin {
    constructor(options) {
      this.prefix = options.dataPrefix;
    }

    'after:HighlightBlock'({block, result}) {
      // ...
    }
  }
```

?

Ha, I didn't even know you could do that.

Reply:
%)  How to invoke this thing?

Reply:
You don't ever invoke it directly yourself, you just write it - the plugin system invokes it as necessary.  I also think it's nice because it differentiates the callbacks from regular internal functions your class might have.

But if you actually wanted to know you just use the `[""]` attribute accessor form:

```
object["weird:key:name"]()
```